apps/aeutil/lib/patricia_merkle_tree.ex

```diff
@@ -0,0 +1,97 @@
+defmodule Aeutil.PatriciaMerkleTree do
+  @moduledoc """
+
+  This module provides apis for creating, updating, deleting
+  patricia merkle tries, The actual handler is https://github.com/exthereum/merkle_patricia_tree
+
+  """
+
+  alias MerklePatriciaTree.Trie
+  alias MerklePatriciaTree.Proof
+  alias MerklePatriciaTree.DB.ExternalDB
+
+  alias Aecore.Persistence.Worker, as: Persistence
+
+  @typedoc """
+  Depending on the name, different data base ref will
+  be used for the trie creaton.
+  """
+  @type trie_name :: :account | :txs | :proof
+
+  @spec root_hash(Trie.t()) :: binary
+  def root_hash(%{root_hash: root_hash}), do: root_hash
+
+  @doc """
+  Creating new trie.
+  """
+  @spec new(trie_name) :: Trie.t()
+  def new(trie_name), do: Trie.new(ExternalDB.init(get_db_handlers(trie_name)))
+
+  @doc """
+  Create new trie with specific hash root
+  """
+  @spec new(trie_name, binary) :: Trie.t()
+  def new(trie_name, root_hash) do
+    Trie.new(ExternalDB.init(get_db_handlers(trie_name)), root_hash)
+  end
+
+  @spec new(trie_name) :: Trie.t()
+  defp get_db_handlers(trie_name) do
+    %{put: Persistence.db_handler_put(trie_name), get: Persistence.db_handler_get(trie_name)}
+  end
+
+  @doc """
+  Retrieve value from trie.
+  """
+  @spec lookup(Trie.key(), Trie.t()) :: {:ok, Trie.value()} | :none
+  def lookup(key, trie) do
+    case Trie.get(trie, key) do
+      nil -> :none
+      val -> {:ok, val}
+    end
+  end
+
+  @doc """
+  Retrieve value from trie and construct proof.
+  """
+  @spec lookup_with_proof(Trie.key(), Trie.t()) :: :none | {:ok, Trie.value(), Trie.t()}
+  def lookup_with_proof(key, trie) do
+    case Proof.construct_proof({trie, key, new(:proof)}) do
+      {nil, _proof} -> :none
+      {value, proof} -> {:ok, value, proof}
```

Comment:
You could name `value` as `val` and follow the previous `lookup()` function variable naming.